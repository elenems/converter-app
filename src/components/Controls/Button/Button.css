.MuiButton-label{
    font-size: 24px;
}