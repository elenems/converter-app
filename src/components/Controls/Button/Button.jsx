import React from "react";
import MaterialButton from "@material-ui/core/Button";
import { MdChevronLeft } from "react-icons/md";
import { MdChevronRight } from "react-icons/md";
import Tooltip from "@material-ui/core/Tooltip";
import "./Button.css";

export default React.memo(function Button({ handleClick }) {
  return (
    <Tooltip title="Switch currencies" placement="bottom">
      <MaterialButton onClick={handleClick} variant="contained" color="primary">
        <MdChevronLeft />
        <MdChevronRight />
      </MaterialButton>
    </Tooltip>
  );
});
