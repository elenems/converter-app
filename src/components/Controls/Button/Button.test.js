import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { MdChevronLeft } from "react-icons/md";
import { MdChevronRight } from "react-icons/md";

configure({ adapter: new Adapter() });
describe('Button', () => {

  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Button />).debug();
    expect(component).toMatchSnapshot();
  });

  it('renders left chevron icon', () => {
    const component = shallow(<Button />);
    expect(component.find(MdChevronLeft).exists()).toEqual(true);
  });

  it('renders right chevron icon', () => {
    const component = shallow(<Button />);
    expect(component.find(MdChevronRight).exists()).toEqual(true);
  });
});