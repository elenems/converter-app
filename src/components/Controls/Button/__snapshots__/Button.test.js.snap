// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly in "debug" mode 1`] = `
"<WithStyles(Tooltip) title=\\"Switch currencies\\" placement=\\"bottom\\">
  <WithStyles(ForwardRef(Button)) onClick={[undefined]} variant=\\"contained\\" color=\\"primary\\">
    <MdChevronLeft />
    <MdChevronRight />
  </WithStyles(ForwardRef(Button))>
</WithStyles(Tooltip)>"
`;
